buildstate-maven-extension/src/main/java/io/helidon/build/state/maven/ProjectStateLifecyclePartitipant.java

```diff
@@ -0,0 +1,243 @@
+                | PluginDescriptorParsingException
+                | MojoNotFoundException
+                | InvalidPluginDescriptorException e) {
+            logger.warn("Unable to resolve mojos for phase: " + phase, e);
+            return List.of();
+        }
+    }
+
+    private MojoExecution configureExecution(MojoExecution execution, MavenSession session, MavenProject project) {
+        try {
+            executionPlanCalculator.setupMojoExecution(session, project, execution);
+        } catch (PluginNotFoundException
+                | PluginResolutionException
+                | PluginDescriptorParsingException
+                | MojoNotFoundException
+                | InvalidPluginDescriptorException
+                | NoPluginFoundForPrefixException
+                | LifecyclePhaseNotFoundException
+                | LifecycleNotFoundException
+                | PluginVersionResolutionException ex) {
+            logger.error("Unable to configure mojo: " + execution, ex);
+            return null;
+        }
+        return execution;
+    }
+
+    private void skipExecutions(Plugin plugin, List<RecordedExecution> executions) {
+        // find matching executions for this plugin
+        Map<String, List<String>> goalsById =
+                plugin.getExecutions()
+                      .stream()
+                      .flatMap(exec ->
+                              exec.getGoals()
+                                  .stream()
+                                  .map(goal -> Map.entry(exec.getId(), goal))
+                                  .filter(e -> isRecordedExecution(plugin, e.getValue(), e.getKey(), executions)))
+                      .collect(Collectors.toMap(e -> e.getKey(), e -> List.of(e.getValue()), Helper::listConcat));
+
+        // remove the goals
+        plugin.getExecutions().forEach(pluginExecution -> {
+            String executionId = pluginExecution.getId();
+            goalsById.getOrDefault(executionId, List.of())
+                          .forEach(goal -> {
+                              logger.debug("Skipping: " + plugin.getId() + "@" + executionId);
+                              pluginExecution.removeGoal(goal);
+                          });
+        });
+
+        // remove the executions without goals
+        Iterator<PluginExecution> it = plugin.getExecutions().iterator();
+        while (it.hasNext()) {
+            PluginExecution pluginExecution = it.next();
+            if (pluginExecution.getGoals().isEmpty()) {
+                it.remove();
+            }
+        }
+    }
+
+    private static boolean isRecordedExecution(Plugin plugin,
+                                               String goal,
+                                               String executionId,
+                                               List<RecordedExecution> executions) {
+        return executions
+                .stream()
+                .filter(e -> e.matches(plugin, goal, executionId))
```

Comment:
You can use `anyMatch(test)` to simplify this; no need to change it, just FYI in case you weren't aware.

Reply:
nice tip, I'll use it.